### PR TITLE
Cleanup some cache helpers and remove indirections

### PR DESCRIPTION
**Pull request by markusthoemmes**
Indirections make stuff hard to follow and in this case, those seem unnecessary. So let's drop them.

### CODE REVIEW

File: pkg/generator/caches_test.go [hunk 1]
```diff
@@ -187,7 +187,8 @@ func createTestDataForIngress(
 		},
 	}, &translatedIngress)
 
-	caches.addStatusVirtualHost()
+	statusVirtualHost := statusVHost()
+	caches.statusVirtualHost = &statusVirtualHost
```
Comment (davidor) [inline pkg/generator/caches_test.go:191]: Is this needed? Should be already done in this call, right?
https://github.com/knative-sandbox/net-kourier/pull/307/files#diff-99be96ee4d2233307511fe08afab35fce19f6bb870f9a92d140cee81a2fa3b4aR44

Comment (markusthoemmes) [reply pkg/generator/caches_test.go]: Good call, fixed!

File: pkg/generator/caches.go [hunk 2]
```diff
@@ -47,13 +47,23 @@ type Caches struct {
 }
 
 func NewCaches(ctx context.Context, logger *zap.SugaredLogger, kubernetesClient kubeclient.Interface, extAuthz bool) (*Caches, error) {
+	statusVirtualHost := statusVHost()
 	c := &Caches{
 		translatedIngresses: make(map[string]*translatedIngress),
 		clusters:            newClustersCache(logger.Named("cluster-cache")),
 		clustersToIngress:   make(map[string][]string),
 		logger:              logger,
+		statusVirtualHost:   &statusVirtualHost,
```
Comment (davidor) [inline pkg/generator/caches.go:56]: Given that it's not used anywhere else, we could do `&statusVHost()` here and delete the assignment above.

Comment (markusthoemmes) [reply pkg/generator/caches.go:56]: Nope, that notation doesn't work. If we merge #309 first though, we can do that.

**Review by davidor — APPROVED** [review]
/lgtm
